sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -123,7 +123,11 @@ object JavaTypeInference {
         val beanInfo = Introspector.getBeanInfo(typeToken.getRawType)
         val properties = beanInfo.getPropertyDescriptors.filterNot(_.getName == "class")
```

Comment:
Yes, It seems loosely related with the JIRA here. I am fine with changing it if any committer could confirm. Otherwise, I would like to avoid a behaviour change here.